bundle/Core/Submitter.php

```diff
@@ -79,6 +89,13 @@ private function createSubmission(Form $formEntity, string $user): FormSubmissio
         $formSubmission->setForm($formEntity);
         $formSubmission->setData($data);
 
+        /* @var TokenInterface $token */
+        $token = $this->tokenStorage->getToken();
+        if ($token instanceof UsernamePasswordToken) {
+            $user = $token->getUser()->getAPIUser();
```

Comment:
here you need to check that ` $token->getUser()` will really be the eZ User (I don't remember but the namespace includes Security / eZ and something) only in this situation you will be able to call getAPIUser

Thx!